File: WebApi/WebApi/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Http;
using System.Web.Mvc;
using WebApi.Enums;
using WebApi.Models;

namespace WebApi.Controllers
{
    public class HomeController : ApiController
    {
        // GET: Home
        public ListaMovimientosResponseModel GetListaMovimientosResponseModel(int id)
        {
            ListaMovimientosResponseModel respuestaDto = new ListaMovimientosResponseModel();
            try
            {
                if (id == 1)
                {
                    respuestaDto.CodigoRespuesta = Enums.Enumerados.TipoRespuestaEnum.Correcto;
                    respuestaDto.Mensaje = "Datos home";
                    respuestaDto.Movimientos = new List<TipoMovimientosResponseModel>();
                    for (int i = 0; i < 10; i++)
                    {
                        TipoMovimientosResponseModel movimiento = new TipoMovimientosResponseModel();
                        movimiento.IdMovimiento = i;
                        movimiento.Titulo = "Titulo " + i;
                        movimiento.Descripcion = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec porttitor, quam ut rutrum fermentum, lorem quam volutpat justo, quis gravida nibh risus at tortor. Sed efficitur justo a tortor dapibus, sed facilisis erat convallis. Aliquam sit amet ultricies urna. Donec pharetra urna sed eros commodo, vitae dapibus metus malesuada.";
                        movimiento.Imagen = "http://icons.iconarchive.com/icons/igh0zt/ios7-style-metro-ui/128/MetroUI-Other-Task-icon.png";
                        respuestaDto.Movimientos.Add(movimiento);
                        if (i < 3)
                            movimiento.TipoMovimiento = Enumerados.TipoMovimientoEnum.Agenda;
                        else if (i < 7)
                            movimiento.TipoMovimiento = Enumerados.TipoMovimientoEnum.Finanzas;
                        else if (i < 10)
                            movimiento.TipoMovimiento = Enumerados.TipoMovimientoEnum.Tareas;
                    }
                }
                else
                {
                    respuestaDto.CodigoRespuesta = Enums.Enumerados.TipoRespuestaEnum.Error;
                    respuestaDto.Mensaje = "Número inválido";
                }
            }
            catch (Exception ex)
            {
                respuestaDto.CodigoRespuesta = Enums.Enumerados.TipoRespuestaEnum.Excepcion;
                respuestaDto.Mensaje = ex.Message;
            }
            return respuestaDto;
        }
    }
}

File: RDAsistente/app/src/main/java/mx/reddam/rdasistente/Negocio/Controllers/HomeController.java
package mx.reddam.rdasistente.Negocio.Controllers;

import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import mx.reddam.rdasistente.Enums.EnumTipoMovimiento;
import mx.reddam.rdasistente.Enums.EnumTipoRespuesta;
import mx.reddam.rdasistente.Models.ListaMovimientosModel;
import mx.reddam.rdasistente.Models.TiposMovimientosModel;

/**
 * Created by Rogelio Andrade on 08/08/2018.
 */

public class HomeController {
    public static HomeController instance = null;

    public static HomeController getInstance(){
        Object lock = new Object();
        if(instance==null){
            synchronized (lock){
                if(instance == null){
                    return new HomeController();
                }else{
                    return instance;
                }
            }
        }else {
            return instance;
        }
    }

    public ListaMovimientosModel fillData(){
        ListaMovimientosModel listaModel = new ListaMovimientosModel();
        try{
            listaModel.setCodigoRespuesta(EnumTipoRespuesta.Correcto);
            listaModel.setMensaje("Datos home");
            List<TiposMovimientosModel> movimientosLista = new ArrayList<>();
            for (int i = 0; i < 10; i++)
            {
                TiposMovimientosModel movimiento = new TiposMovimientosModel();
                movimiento.setIdMovimiento(i);
                movimiento.setTitulo("Titulo ".concat(Integer.toString(i)));
                movimiento.setDescripcion("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec porttitor, quam ut rutrum fermentum, lorem quam volutpat justo, quis gravida nibh risus at tortor. Sed efficitur justo a tortor dapibus, sed facilisis erat convallis. Aliquam sit amet ultricies urna. Donec pharetra urna sed eros commodo, vitae dapibus metus malesuada.");
                movimiento.setImagen("http://i.imgur.com/DvpvklR.png");
                movimientosLista.add(movimiento);
                if (i < 3)
                    movimiento.setTipoMovimiento(EnumTipoMovimiento.Agenda);
                else if (i < 7)
                    movimiento.setTipoMovimiento(EnumTipoMovimiento.Finanzas);
                else if (i < 10)
                    movimiento.setTipoMovimiento(EnumTipoMovimiento.Tareas);
            }
            listaModel.setMovimientos(movimientosLista);
        }catch (Exception ex){
            Log.e("Error en HomeController", "fillData: "+ex.getMessage());
        }
        return listaModel;
    }
    private HomeController(){}

}
